scheduler/cmd/scheduler/main.go

```diff
@@ -279,10 +278,9 @@ func main() {
 		logger.WithError(err).Fatal("Failed to load Kafka config")
 	}
 
-	numPartitions, err := strconv.Atoi(kafkaConfigMap.Topics["numPartitions"].(string))
+	numPartitions, err := kafkaConfigMap.GetNumPartitions()
```

Comment:
A note that in the new setup (the PR that I'm working on) the number of partitions are no longer fetched here. However, we might be able to reuse the other changes in components/kafka if pipeline-gw/model-gw require the number of partitions.